### PR TITLE
[Program:GCI] Created cron job to remove unverified users

### PR DESCRIPTION
**Pull request by rudrakshjhalanii07**
### Description
Created a cron job function to remove unverified users. The time limit has been set to 24 hours itself, as after that the verification token gets expired, so to sync with it.

Fixes # [ISSUE]
#115 

### Type of Change:
**Delete irrelevant options.**

- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested
I added test cases and all the test cases ran successfully.

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: templates/email_confirmation.html [hunk 1]
```diff
@@ -6,6 +6,8 @@
     Before you get started, we need to verify your email:</p>
 <br>
 <p><a href="{{ confirm_url }}">{{ confirm_url }}</a></p>
+<p>This token will expire in {{ expiry_time }} hours.
```
Comment (isabelcosta) [inline templates/email_confirmation.html:9]: Great idea :)

**Review by isabelcosta — APPROVED** [review]
Great job @Hardik0403 🎉 
Thank you for your contribution!

File: templates/email_confirmation.html (hunk 1 above)
Comment (rudrakshjhalanii07) [reply templates/email_confirmation.html:9]: Thank you:)😊

File: app/schedulers/remove_unverified_user_cron_job.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    from run import application
+    with application.app_context():
+        from app.database.models.user import UserModel
+        all_users = UserModel.query.all()
+
+        current_time = time.time()
+
+        for user in all_users:
+            # The date of removing unverified user, after the expiry of verification token.
+            token_expiry_time = user.registration_date + EMAIL_VERIFICATION_TOKEN_TIME_TO_EXPIRE
```
Comment (annabauza) [inline app/schedulers/remove_unverified_user_cron_job.py:20]: how long is that?

File: app/schedulers/remove_unverified_user_cron_job.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    """
+    This function iterates over all the users and
+    checks if a user has unverified email and
+    the verification token of the user has expired.
+    if True then the user is removed, if False does nothing.
+    """
+    from run import application
+    with application.app_context():
+        from app.database.models.user import UserModel
+        all_users = UserModel.query.all()
```
Comment (annabauza) [inline app/schedulers/remove_unverified_user_cron_job.py:14]: filter by unverified please

**Review by annabauza — CHANGES_REQUESTED** [review]
Please apply changes I ask for. Thank you!

File: app/schedulers/remove_unverified_user_cron_job.py (hunk 2 above)
Comment (rudrakshjhalanii07) [reply app/schedulers/remove_unverified_user_cron_job.py:20]: 24 hours, as soon as the verification token expires..

File: app/schedulers/remove_unverified_user_cron_job.py (hunk 3 above)
Comment (rudrakshjhalanii07) [reply app/schedulers/remove_unverified_user_cron_job.py:14]: But doing so will decrease the efficiency of the code. If the filter function is applied over here, then internally a loop would run to filter the list, and then another loop will run to remove the unverified users. Currently this work is done only by a single loop.

**Comment by annabauza** [comment]
closing due to merge #350

**Pull request closed without merging**